asf_heat_pump_suitability/pipeline/prepare_features/conservation_areas.py

```diff
@@ -24,56 +34,55 @@ def generate_df_conservation_area_data_availability(
+    Generate dataframe of UK local authority districts (LADs) with indicator of building conservation area data
+    availability.
 
     Args:
         ladcd_col (str): name of column in local authority district (LAD) boundaries file with LAD codes
 
     Returns:
-        pl.DataFrame: conservation area data availability per LAD in the UK
+        pl.DataFrame: building conservation area data availability per LAD in the UK
     """
-    cons_areas_gdf = transform_gdf_conservation_areas_england()
-    council_bounds = get_datasets.load_gdf_ons_council_bounds().to_crs(epsg="27700")
+    cons_areas_gdf = transform_gdf_building_cons_areas()
+    council_bounds = get_datasets.load_gdf_ons_council_bounds()
 
     # Join conservation areas to their councils
     df = council_bounds.sjoin(cons_areas_gdf, how="left", predicate="intersects")[
-        [ladcd_col, "name"]
+        [ladcd_col, "sitename"]
     ].replace("No data available for publication by HE", np.nan)
 
-    df = df.groupby(ladcd_col).agg({"name": "count"})
-    df["lad_conservation_area_data_available"] = df["name"].astype(bool)
-    df = df.drop(columns=["name"]).reset_index()
+    df = df.groupby(ladcd_col).agg({"sitename": "count"})
+    df["lad_conservation_area_data_available"] = df["sitename"].astype(bool)
+    df = df.drop(columns=["sitename"]).reset_index()
 
     return pl.from_pandas(df)
 
 
-def generate_df_uprn_to_cons_area(epc_df: pl.DataFrame) -> pl.DataFrame:
+def generate_df_uprn_to_cons_area(df: pl.DataFrame) -> pl.DataFrame:
     """
-    Generate dataframe of UPRNs located inside conservation areas.
+    Generate dataframe of UPRNs located within building conservation areas in England and Wales.
 
     Args:
-        epc_df (pl.DataFrame): EPC dataset with UPRN column and X and Y coordinate columns in BNG
+        df (pl.DataFrame): EPC dataset with UPRN column and X and Y coordinate columns in BNG
 
     Returns:
-        pl.DataFrame: dataframe of EPC UPRNs in conservation areas
+        pl.DataFrame: dataframe of EPC UPRNs in building conservation areas in England and Wales
     """
     # Convert BNG x, y coordinates to point geometries
-    epc_df = lat_lon.generate_gdf_uprn_coords(epc_df)[["UPRN", "geometry"]]
+    df = lat_lon.generate_gdf_uprn_coords(df)[["UPRN", "geometry"]]
 
-    # Load conservation areas England
-    cons_areas_gdf = transform_gdf_conservation_areas_england()
+    # Load and transform conservation areas in England and Wales
+    cons_areas_gdf = transform_gdf_building_cons_areas()
 
     # Join EPC UPRNs within or on boundaries of conservation areas
-    epc_df = epc_df.sjoin(cons_areas_gdf, how="inner", predicate="intersects").drop(
-        columns=["index_right", "geometry"]
+    df = (
+        df.sjoin(cons_areas_gdf, how="inner", predicate="intersects")
+        .drop(columns=["index_right", "geometry"])
+        .drop_duplicates(subset="UPRN")
     )
```

Comment:
Thanks Liz! Great idea :) 
Yeh loading the lat lon coordinates dataset takes ages because it is an absolutely huge dataset. It has all UPRNs for whole of GB with their latitude and longitude...

Reply:
I'll see if we're loading additional unnecessary columns